feat: [I4] setup project boilerplate

NickolasKemp opened the pull request.

elly-allexa committed:
Add src and docs folders with placeholder files

elly-allexa committed:
Add empty src folder with .gitkeep placeholder

elly-allexa committed:
Add documentation structure with placeholders in devdoc branch

elly-allexa committed:
Merge pull request #3 from SEBPEO/devdoc

Add documentation structure with placeholders in devdoc branch

NickolasKemp committed:
[I4] project init

et56137-fer commented on frontend/src/App.tsx:
```diff
@@ -0,0 +1,35 @@
+  return (
+    <>
+      <div>
+        <a href="https://vite.dev" target="_blank">
+          <img src={viteLogo} className="logo" alt="Vite logo" />
+        </a>
+        <a href="https://react.dev" target="_blank">
```
https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/rel

noopener is not needed alongside noreferrer

et56137-fer reviewed (changes requested):
check the comments please

elly-allexa committed:
Update .gitignore to include additional patterns

Updating the file for :
Secrets leak prevention: .env (no API keys or passwords go public!)
Gigantic repo size: node_modules/, dist/
Local junk: .DS_Store, .log, *.tmp
Database safety: prevents your local database file (if SQLite will be used) from being uploaded
Editor clutter: .vscode, .idea, etc.

elly-allexa committed:
Refactor .gitignore to consolidate ignored files

Fixed a mistake connected to the files confusion

elly-allexa committed:
Delete src/.gitkeep

NickolasKemp commented on backend/package.json:
```diff
@@ -0,0 +1,73 @@
+  "description": "",
+  "author": "",
+  "private": true,
+  "license": "UNLICENSED",
```
good comment, but I think it does not matter for us

we are not an open-source project)

NickolasKemp commented on backend/.gitignore:
```diff
@@ -0,0 +1,56 @@
+/dist
+/node_modules
+/build
+
```
no, it's fine
